lib/rules/lint-attribute-indentation.js

```diff
@@ -0,0 +1,301 @@
+                  as |employee address|}}
+            {{employee.name}}
+          {{/employee-details}}
+          */
+          actual = {
+            line: programStartLoc.line,
+            column: getWhiteSpaceLength(this.source[programStartLoc.line - 1])
+          };
+        }
+      }
+    }
+    return {
+      actual,
+      expected
+    };
+  }
+
+
+  validateBlockParams(node) {
+    /*
+      Validates alignment of the block params.
+
+      {{#employee-details
+        employeeId=employeeId
+      as |name age address|}}
+        {{name}}, {{age}}, {{address}}
+      {{/employee-details}}
+    */
+
+    let location = this.getBlockParamStartLoc(node);
+    let actual = location.actual;
+    let expected = location.expected;
+
+    if (actual.line !== expected.line ||
+      actual.column !== expected.column) {
+
+      let blockParamStatement = this.sourceForNode({ loc: { start: actual, end: node.program.loc.start } }).trim();
+
+      let message = `Incorrect indentation of block params '${blockParamStatement}' beginning at L${actual.line}:C${actual.column}. Expecting the block params to be at L${expected.line}:C${expected.column} with an indentation of ${expected.column} but was found at ${actual.column}.`;
+
+      this.log({
+        message,
+        line: actual.line,
+        column: actual.column,
+        source: this.sourceForNode(node)
+      });
+    }
+  }
+
+  iterateParams(params, type, expectedLineStart, expectedColumnStart, node) {
+    let paramType, namePath;
+
+    if (type === 'positional') {
+      paramType = 'positional param';
+      namePath = 'original';
+    } else {
+      paramType = 'attribute';
+      namePath = 'key';
+    }
+    params.forEach((param) => {
+      let actualStartLocation = param.loc.start;
+      if (expectedLineStart !== actualStartLocation.line || expectedColumnStart !== actualStartLocation.column) {
+        let paramName = param[namePath];
+        let message = `Incorrect indentation of ${paramType} '${paramName}' beginning at L${actualStartLocation.line}:C${actualStartLocation.column}. Expected '${paramName}' to be indentation at an of ${expectedColumnStart} but was found at ${actualStartLocation.column}`;
+        this.log({
+          message,
+          line: actualStartLocation.line,
+          column: actualStartLocation.column,
+          source: this.sourceForNode(node)
+        });
+      }
+      expectedLineStart++;
+    });
+  }
+
+  validateParamsAndHashPairs (node) {
+    /*
+        Validates both the positional and named params for both block and non-block form.
+
+        {{contact-details
+          age
+          firstName=firstName
+          fullName=fullName
+        }}
+    */
+    let expectedColumnStart = node.loc.start.column + 2; //params should be after 2 positions from component start node
+    let expectedLineStart = node.loc.start.line + 1;
+
+    this.iterateParams(node.params, 'positional', expectedLineStart, expectedColumnStart, node);
```

Comment:
Current implementation enforces the `positional params` to follow the same indentation rule.

Reply:
Seems good to me (I think)....